applications/tari_base_node/src/command_handler.rs

```diff
@@ -376,11 +376,50 @@ impl CommandHandler {
     }
 
     /// Function to process the get-mempool-state command
-    pub fn get_mempool_state(&self) {
+    pub fn get_mempool_state(&self, full: bool, filter: Option<String>) {
```

Comment:
Maybe instead of passing in `full`, you could show the transaction body if filter is Some and the Display impl of StateResponse if filter is None

Reply:
Reasoning, we probably dont want to display all of the transaction bodies as that will just scroll off the screen - only one at a time